src/algebra/big_operators/order.lean

```diff
@@ -241,6 +241,39 @@ mul_card_image_le_card_of_maps_to (λ x, mem_image_of_mem _) n hn
 
 end pigeonhole
 
+section double_counting
+variables [decidable_eq α] {s : finset α} {B : finset (finset α)} {n : ℕ}
+
+/-- If every element belongs to at most `n` finsets, then the sum of their sizes is at most `n`
+times how many they are. -/
+lemma sum_card_inter_le (h : ∀ a ∈ s, (B.filter $ (∈) a).card ≤ n) :
+  ∑ t in B, (s ∩ t).card ≤ s.card * n :=
+begin
+  simp_rw [←filter_mem_eq_inter, card_eq_sum_ones, sum_filter],
+  rw [sum_comm, ←card_eq_sum_ones],
```

Comment:
Can you get there any faster with `sum_const`?

Reply:
I'm thinking maybe `refine (sum_le_of_forall_le h).trans _` would get you there quicker.

Reply:
Not really. But I managed to get rid of this `←card_eq_sum_ones` which was just here to cancel a too eager rewrite.